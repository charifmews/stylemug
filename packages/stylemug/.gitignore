compiler.js
dist/*
!dist/index.js