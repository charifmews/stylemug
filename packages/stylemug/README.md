# stylemug

A fast css-in-js library that extracts atomic CSS rules to a .css file.

For more information, visit https://github.com/matvp91/stylemug
