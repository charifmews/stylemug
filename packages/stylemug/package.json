{
  "name": "stylemug",
  "version": "1.0.1",
  "main": "dist/runtime.js",
  "module": "src/index.js",
  "license": "MIT",
  "dependencies": {},
  "scripts": {
    "build": "webpack --mode production"
  },
  "devDependencies": {}
}
