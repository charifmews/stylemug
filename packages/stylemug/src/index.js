export default {
  create(schema) {
    const resolver = (...classNames) => {
      return Object.values(
        Object.assign(
          {},
          ...classNames.map((className) => {
            // When composed from elsewhere, it'll contain the rules directly,
            // as the result of resolver.className.
            if (typeof className === 'object') {
              return className;
            }
            return schema[className];
          })
        )
      ).join(' ');
    };

    for (let key in schema) {
      resolver[key] = schema[key];
    }

    return resolver;
  },
};
