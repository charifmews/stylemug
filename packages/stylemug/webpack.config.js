const path = require('path');

module.exports = [
  {
    entry: './src/index.js',
    output: {
      libraryTarget: 'commonjs2',
      filename: 'runtime.js',
      path: path.resolve(__dirname, 'dist'),
    },
    module: {
      rules: [
        {
          test: /\.m?js$/,
          exclude: /(node_modules|bower_components)/,
          use: 'babel-loader',
        },
      ],
    },
  },
  {
    entry: 'stylemug-compiler',
    target: 'node',
    output: {
      libraryTarget: 'commonjs2',
      filename: 'compiler.js',
      path: __dirname,
    },
  },
];
