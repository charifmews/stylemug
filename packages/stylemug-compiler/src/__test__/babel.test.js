const babel = require('@babel/core');
const plugin = require('../babel');

jest.mock('../compile', () => ({
  compileSchema: () => ({
    className: {
      hash: {
        keyId: 'id',
      },
    },
  }),
}));

describe('babel plugin', () => {
  it('should replace create argument', () => {
    const example = `
      import stylemug from 'stylemug';

      const styles = stylemug.create({
        default: {
          color: 'red',
        },
      });
    `;

    const { code } = babel.transform(example, {
      plugins: [plugin],
    });

    expect(code).toMatchSnapshot();
  });
});
