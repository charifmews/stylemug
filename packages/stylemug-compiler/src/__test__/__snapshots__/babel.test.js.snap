// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel plugin should replace create argument 1`] = `
"import stylemug from 'stylemug';
const styles = stylemug.create({
  className: {
    id: \\"hash\\"
  }
});"
`;
